### PR TITLE
Improve header 10

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Part of a waterfall PR, the final one being #596 
- General styling
- Defines a general constant for the main menu content in `mainMenu.ts`. A single file which can be used to define the main menu as a while.
- Define a `const getMainMenu` structure with conditional logic supporting the mainMenu.ts constant data structure.
- Added a util function `toggleBodyClass.ts` to toggle the `document.body` class, with the class provided as an argument. Used to toggle the 'noScroll` class whenever an overlay (like the main menu on mobile) is active.

### CODE REVIEW

File: src/custom/components/Header/index.tsx [hunk 1]
```diff
@@ -88,17 +76,74 @@ export default function Header() {
...
-    isUpToLarge && isMobileMenuOpen
-      ? document.body.classList.add('noScroll')
-      : document.body.classList.remove('noScroll')
+    isOrdersPanelOpen ? toggleBodyClass('noScroll', true) : toggleBodyClass('noScroll', false)
+    isUpToLarge && isMobileMenuOpen ? toggleBodyClass('noScroll', true) : toggleBodyClass('noScroll', false)
   }, [isOrdersPanelOpen, isMobileMenuOpen, isUpToLarge])
 
+  const getMainMenu = useMemo(
+    () =>
+      MAIN_MENU.map(({ title, url, externalURL, items }: MAIN_MENU_TYPE, index) =>
```
Comment (alfetopito) [inline src/custom/components/Header/index.tsx:93]: I'm sure @anxolin will love this part 😅 

Anyway, would you be able to simplify this a bit?
Maybe use intermediate components or callbacks to reduce a bit the conditional chaining

**Review by alfetopito — APPROVED** [review]

File: src/custom/components/Header/index.tsx (hunk 1 above)
Comment (fairlighteth) [reply src/custom/components/Header/index.tsx:93]: @alfetopito If we can simplify it further, that'd be great. Would any of you mind checking PR https://github.com/cowprotocol/cowswap/pull/596 (the current last one) and suggest how to further simplify? I know I did some changes up to the last PR.

File: src/custom/components/AccountDetails/Transaction/CancelationModal.tsx [hunk 2]
```diff
@@ -48,7 +49,7 @@ function RequestCancellationModal(props: RequestCancellationModalProps): JSX.Ele
               <p>
                 This means that a solver might already have included the order in a solution even if this cancellation
                 is successful. Read more in the{' '}
-                <a target="_blank" href="/#/faq/trading#can-i-cancel-an-order">
+                <a target="_blank" rel="noreferrer" href={`${Routes.FAQ_TRADING}#can-i-cancel-an-order`}>
```
Comment (anxolin) [inline src/custom/components/AccountDetails/Transaction/CancelationModal.tsx:52]: I think we usually create the links using <StyledInternalLink /> <StyledLink /> or <ExternalLink

is not part of this PR, but just curious if we know why this is like this?

File: src/custom/components/Header/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/custom/components/Header/index.tsx:94]: Yes, this is way too complex...

we can break it up a bit into smaller pieces. A map inside another map and with all these ternary operators are big red 🚩

File: src/custom/constants/mainMenu.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import IMAGE_DOCS from 'assets/cow-swap/doc.svg'
+import IMAGE_INFO from 'assets/cow-swap/info.svg'
+import IMAGE_CODE from 'assets/cow-swap/code.svg'
+import IMAGE_DISCORD from 'assets/cow-swap/discord.svg'
+import IMAGE_TWITTER from 'assets/cow-swap/twitter.svg'
+import IMAGE_PIE from 'assets/cow-swap/pie.svg'
+import IMAGE_SLICER from 'assets/cow-swap/ninja-cow.png'
+import IMAGE_GAME from 'assets/cow-swap/game.gif'
+
+export interface MAIN_MENU_TYPE {
```
Comment (anxolin) [inline src/custom/constants/mainMenu.ts:14]: I wonder if these constants should be instead near the menu component, instead of in the general constants

File: src/custom/constants/routes.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// ENUM with routes
+export enum Routes {
```
Comment (anxolin) [inline src/custom/constants/routes.ts:2]: @fairlighteth this I like 👍
we should make sure the all our links start to use this enums. Maybe we can even create a Link which only accepts this enum as the path parameter

**Review by anxolin — COMMENTED** [review]
Lots of things in this PR :)

File: src/custom/components/AccountDetails/Transaction/CancelationModal.tsx (hunk 2 above)
Comment (fairlighteth) [reply src/custom/components/AccountDetails/Transaction/CancelationModal.tsx:52]: Yeah would need to change for sure. In this case I had simply changed the URL but should use the component 👍🏼

File: src/custom/constants/mainMenu.ts (hunk 3 above)
Comment (fairlighteth) [reply src/custom/constants/mainMenu.ts:14]: In this case the icons are specific to the menu tree structure, which comes from the mainMenu types. So my thinking is it should be closely tied to the menu structure. WDYT?

File: src/custom/components/AccountDetails/Transaction/CancelationModal.tsx (hunk 2 above)
Comment (fairlighteth) [reply src/custom/components/AccountDetails/Transaction/CancelationModal.tsx:52]: Just tested this one out when the cancellation modal is open:
- using the <Link> component it opens the page below, which is not great.
- using the <ExternalLink> component I tried not to specify the root domain, but the URL gets malformed: `http://localhost:3000/faq/trading#/can-i-cancel-an-order?chain=rinkeby` 

thinking to leave as is for now in this iteration
